jme3-examples/src/main/java/jme3test/terrain/PBRTerrainAdvancedTest.java

```diff
@@ -0,0 +1,428 @@
+
+
+        matTerrain.setInt("AlbedoMap_5", 5);
+        matTerrain.setFloat("AlbedoMap_5_scale", marbleScale);
+        matTerrain.setFloat("Roughness_5", 1);
+        matTerrain.setFloat("Metallic_5", 0.2f);
+        //matTerrain.setInt("AfflictionMode_5", 0);
+
+        matTerrain.setInt("AlbedoMap_6", 6);
+        matTerrain.setFloat("AlbedoMap_6_scale", gravelScale);
+        matTerrain.setFloat("Roughness_6", 1);
+        matTerrain.setFloat("Metallic_6", 0.01f);
+        //matTerrain.setInt("AfflictionMode_6", 0);
+
+        
+        
+        // NORMAL MAPS  //(int being passed to shader corresponds to the index of the texture's image in the List of images used to create normalParallaxTextureArray)
+      
+        matTerrain.setInt("NormalMap_0", 0);
+        matTerrain.setInt("NormalMap_1", 1);  
+        matTerrain.setInt("NormalMap_2", 2); 
+        matTerrain.setInt("NormalMap_3", 3); 
+        matTerrain.setInt("NormalMap_4", 4); 
+        matTerrain.setInt("NormalMap_5", 5);
+        matTerrain.setInt("NormalMap_6", 6);
+                
+                
+        //METALLIC/ROUGHNESS/AO/EI 
+        matTerrain.setInt("MetallicRoughnessMap_0", 0);
+        matTerrain.setInt("MetallicRoughnessMap_1", 1);  
+        matTerrain.setInt("MetallicRoughnessMap_2", 2); 
+        matTerrain.setInt("MetallicRoughnessMap_3", 3); 
+        matTerrain.setInt("MetallicRoughnessMap_4", 4); 
+        matTerrain.setInt("MetallicRoughnessMap_5", 5);
+        matTerrain.setInt("MetallicRoughnessMap_6", 6);
+        
+        
+        //EMISSIVE
+//        matTerrain.setColor("EmissiveColor_5", ColorRGBA.Blue);
+//        matTerrain.setColor("EmissiveColor_6", ColorRGBA.Red); //these two texture slots (marble & tiledRoad) have packed MRAoEi maps with an emissiveTexture packed into the alpha channel
+//        
+//       
+//        matTerrain.setColor("EmissiveColor_1", new ColorRGBA(0.08f, 0.01f, 0.1f, 0.4f)); //this texture slot does not have a unique emissiveIntensityMap packed into its MRAoEi map, so setting an emissiveColor will apply equal intensity to every pixel
+        
+
+        // CREATE HEIGHTMAP
+        AbstractHeightMap heightmap = null;
+        try {
+            heightmap = new ImageBasedHeightMap(heightMapImage.getImage(), 0.3f);
+            heightmap.load();
+            heightmap.smooth(0.9f, 1);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        
+        terrain = new TerrainQuad("terrain", 65, 513, heightmap.getHeightMap());//, new LodPerspectiveCalculatorFactory(getCamera(), 4)); // add this in to see it use entropy for LOD calculations
+   //     TerrainLodControl control = new TerrainLodControl(terrain, getCamera());
+  //      control.setLodCalculator(new DistanceLodCalculator(65, 2.7f)); // patch size, and a multiplier
+     //   terrain.addControl(control);
+        terrain.setMaterial(matTerrain);
+        terrain.setModelBound(new BoundingBox());
```

Comment:
Why do you set the model bound and then update it? Doesn't `new TerrainQuad` yield a valid bound?

Reply:
I'm not sure actually, but I think you are right that it is unnecessary. Some parts of this test case were not written by me (this part is one of them) and in this case I left this line of code there in case it was necessary for the way the terrain was generated from a heightmap. 'm fairly certain I should be able to remove that line without breaking anything, but I will double check in my local copy of the test case and then remove the setModelBound() call from both test cases